Week6/app/src/main/java/com/em4n0101/mymovies/MoviesFragment.kt

```diff
@@ -36,25 +41,27 @@ class MoviesFragment : Fragment(R.layout.fragment_movies) {
             } else {
                 recyclerViewMovies.layoutManager = GridLayoutManager(it, 4)
             }
-            addObserver()
+
+            getListOfMovies()
         }
     }
 
-    private fun addObserver() {
-        /**
-         * Set the fragment as the observer for changes in the list of movies, if the list is empty
-         * add the movies available to the database
-         */
-        val observer = Observer<List<Movie>> {
-            if (it != null && it.isNotEmpty()) {
-                val adapter = MoviesAdapter(::listItemPressed)
-                adapter.setData(it)
-                recyclerViewMovies.adapter = adapter
-            } else {
-                populateDatabase()
+    private fun getListOfMovies() {
+        lifecycleScope.launch {
+            moviesViewModel.getMoviesFlow().collect { listOfMovies ->
+                if (listOfMovies.isNotEmpty()) {
+                    updateUiWithMovieList(listOfMovies)
+                } else {
+                    populateDatabase()
+                }
```

Comment:
You should collect the flow in viewmodel and then expose it as LiveData value which the Fragment can observe. 

You don't want the view to know about flow and collecting data.